File: lab-1/lab1_2.js
class Animal {
  makeSound() {
    throw new Error(
      "Метод makeSound() повинен бути реалізований у похідному класі."
    );
  }

  sleep() {
    console.log(`${this.constructor.name} is sleeping...`);
  }
}

class Dog extends Animal {
  #name;
  #age;

  constructor(name, age) {
    super();
    this.#name = name;
    this.#age = age;
  }

  makeSound() {
    console.log("Woof! Woof!");
  }

  getName() {
    return this.#name;
  }

  setName(name) {
    this.#name = name;
  }

  getAge() {
    return this.#age;
  }

  setAge(age) {
    this.#age = age;
  }
}

const dog = new Dog("Rex", 5);

console.log(dog.getName()); 
dog.setName("Buddy");
console.log(dog.getName()); 

dog.makeSound();

dog.sleep();

File: lab-1/lab1-2.py
from abc import ABC, abstractmethod

class Animal(ABC):
  @abstractmethod
  def make_sound(self):
    pass

  def sleep(self):
    print(f"{self.__class__.__name__} is sleeping...")

class Dog(Animal):
  def __init__(self, name, age):
    self.__name = name 
    self.__age = age   

  def make_sound(self):
    print("Woof! Woof!")

  def get_name(self):
    return self.__name

  def set_name(self, name):
    self.__name = name

  def get_age(self):
    return self.__age

  def set_age(self, age):
    self.__age = age


# Тестування
dog = Dog("Rex", 5)

print(dog.get_name())  
dog.set_name("Buddy")
print(dog.get_name())  

dog.make_sound() 

dog.sleep() 

